cmd/scollector/collectors/procstats_linux.go

```diff
@@ -134,6 +142,12 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
 					tags["cpu"] = tag_cpu
 				}
 				Add(&md, "linux.cpu"+metric_percpu, value, tags, metadata.Counter, metadata.CHz, cpu_stat_desc[CPU_FIELDS[i]])
+
+				vf, err := strconv.ParseFloat(value, 64)
+				if err != nil {
+					return nil
+				}
+				Add(&md, "linux.percent_cpu"+metric_percpu, vf/total_cpu*100.0, tags, metadata.Counter, metadata.Pct, cpu_stat_desc[CPU_FIELDS[i]])
```

Comment:
OpenTSDB does this confession at query time, what time series database are
you using?
On Jul 6, 2015 19:25, "Peter Fern" <notifications@github.com> wrote:

> In cmd/scollector/collectors/procstats_linux.go
> <https://github.com/bosun-monitor/bosun/pull/1138#discussion_r33993967>:
>
> > @@ -134,6 +142,12 @@ func c_procstats_linux() (opentsdb.MultiDataPoint, error) {
> >  					tags["cpu"] = tag_cpu
> >  				}
> >  				Add(&md, "linux.cpu"+metric_percpu, value, tags, metadata.Counter, metadata.CHz, cpu_stat_desc[CPU_FIELDS[i]])
> > +
> > +				vf, err := strconv.ParseFloat(value, 64)
> > +				if err != nil {
> > +					return nil
> > +				}
> > +				Add(&md, "linux.percent_cpu"+metric_percpu, vf/total_cpu*100.0, tags, metadata.Counter, metadata.Pct, cpu_stat_desc[CPU_FIELDS[i]])
>
> Making the counter values useful would mean performing these calculations
> on client side, and I'm struggling to see how you would actually get there,
> but it's entirely possible I'm missing something?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bosun-monitor/bosun/pull/1138/files#r33993967>.
>